File: examples/php/config.example.php
<?php
/**
 * Contoh File Konfigurasi WACloud API
 * 
 * Copy file ini menjadi config.php dan isi dengan API Key Anda
 * 
 * JANGAN commit file config.php ke repository!
 */

// API Key Anda dari dashboard
define('API_KEY', 'YOUR_API_KEY');

// Base URL API
define('BASE_URL', 'https://app.wacloud.id/api/v1');

/**
 * Helper function untuk membuat HTTP request
 */
function makeRequest($method, $endpoint, $data = null) {
    $url = BASE_URL . $endpoint;
    
    $ch = curl_init($url);
    curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
    curl_setopt($ch, CURLOPT_HTTPHEADER, [
        'X-Api-Key: ' . API_KEY,
        'Content-Type: application/json'
    ]);
    
    switch (strtoupper($method)) {
        case 'GET':
            if ($data) {
                $url .= '?' . http_build_query($data);
                curl_setopt($ch, CURLOPT_URL, $url);
            }
            break;
            
        case 'POST':
            curl_setopt($ch, CURLOPT_POST, true);
            if ($data) {
                curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($data));
            }
            break;
            
        case 'PUT':
            curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'PUT');
            if ($data) {
                curl_setopt($ch, CURLOPT_POSTFIELDS, json_encode($data));
            }
            break;
            
        case 'DELETE':
            curl_setopt($ch, CURLOPT_CUSTOMREQUEST, 'DELETE');
            break;
    }
    
    $response = curl_exec($ch);
    $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);
    $error = curl_error($ch);
    curl_close($ch);
    
    if ($error) {
        return [
            'success' => false,
            'error' => $error
        ];
    }
    
    $result = json_decode($response, true);
    $result['http_code'] = $httpCode;
    
    return $result;
}

/**
 * Helper function untuk menampilkan response
 */
function displayResponse($response) {
    echo "HTTP Code: " . ($response['http_code'] ?? 'N/A') . "\n";
    echo "Success: " . ($response['success'] ? 'Yes' : 'No') . "\n";
    
    if (isset($response['message'])) {
        echo "Message: " . $response['message'] . "\n";
    }
    
    if (isset($response['data'])) {
        echo "Data:\n";
        print_r($response['data']);
    }
    
    if (isset($response['error'])) {
        echo "Error: " . $response['error'] . "\n";
    }
    
    echo "\n" . str_repeat('-', 50) . "\n\n";
}

File: examples/golang/config.example.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"net/http"
	"strings"
)

// Contoh File Konfigurasi WACloud API
// Copy file ini menjadi config.go dan isi dengan API Key Anda
// JANGAN commit file config.go ke repository!

const (
	// API Key Anda dari dashboard
	API_KEY = "YOUR_API_KEY"
	// Base URL API
	BASE_URL = "https://app.wacloud.id/api/v1"
)

// Response structure
type APIResponse struct {
	Success  bool                   `json:"success"`
	HTTPCode int                    `json:"http_code"`
	Data     map[string]interface{} `json:"data"`
	Message  string                 `json:"message"`
	Error    string                 `json:"error"`
}

// MakeRequest membuat HTTP request ke API
func MakeRequest(method, endpoint string, data interface{}) (*APIResponse, error) {
	url := BASE_URL + endpoint

	var body io.Reader
	if data != nil && (method == "POST" || method == "PUT") {
		jsonData, err := json.Marshal(data)
		if err != nil {
			return nil, err
		}
		body = bytes.NewBuffer(jsonData)
	}

	req, err := http.NewRequest(method, url, body)
	if err != nil {
		return nil, err
	}

	req.Header.Set("X-Api-Key", API_KEY)
	req.Header.Set("Content-Type", "application/json")

	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()

	bodyBytes, err := io.ReadAll(resp.Body)
	if err != nil {
		return nil, err
	}

	var apiResp APIResponse
	apiResp.HTTPCode = resp.StatusCode

	if err := json.Unmarshal(bodyBytes, &apiResp); err != nil {
		// Jika response bukan JSON, set error
		apiResp.Success = false
		apiResp.Error = string(bodyBytes)
		return &apiResp, nil
	}

	// Extract data if exists
	var result map[string]interface{}
	if err := json.Unmarshal(bodyBytes, &result); err == nil {
		if data, ok := result["data"].(map[string]interface{}); ok {
			apiResp.Data = data
		}
	}

	return &apiResp, nil
}

// DisplayResponse menampilkan response ke console
func DisplayResponse(resp *APIResponse) {
	fmt.Printf("HTTP Code: %d\n", resp.HTTPCode)
	fmt.Printf("Success: %v\n", resp.Success)

	if resp.Message != "" {
		fmt.Printf("Message: %s\n", resp.Message)
	}

	if resp.Data != nil && len(resp.Data) > 0 {
		fmt.Println("Data:")
		jsonData, _ := json.MarshalIndent(resp.Data, "", "  ")
		fmt.Println(string(jsonData))
	}

	if resp.Error != "" {
		fmt.Printf("Error: %s\n", resp.Error)
	}

	fmt.Println("\n" + strings.Repeat("-", 50) + "\n")
}

